### ISSUE TITLE
Support scala 2.11

### ISSUE THREAD

**Issue by agustafson** [neo4j/neo4j#2881]
Using neo4j 2.1.3 and scala 2.11.2 we get the following when attempting to create a `WrappingNeoServer`. Is there any chance that the cypher dependencies can be cross-built and retrieved for multiple scala versions?

```
[error]    NoClassDefFoundError: : scala/runtime/AbstractPartialFunction$mcVL$sp  (HintVerifier.scala:31)
[error] org.neo4j.cypher.internal.compiler.v2_0.executionplan.verifiers.HintVerifier$.<init>(HintVerifier.scala:31)
[error] org.neo4j.cypher.internal.compiler.v2_0.executionplan.verifiers.HintVerifier$.<clinit>(HintVerifier.scala)
[error] org.neo4j.cypher.internal.compiler.v2_0.CypherCompiler.<init>(CypherCompiler.scala:33)
[error] org.neo4j.cypher.internal.CypherCompiler.<init>(CypherCompiler.scala:62)
[error] org.neo4j.cypher.ExecutionEngine.createCompiler(ExecutionEngine.scala:151)
[error] org.neo4j.cypher.ExecutionEngine.<init>(ExecutionEngine.scala:49)
[error] org.neo4j.cypher.javacompat.ExecutionEngine.createInnerEngine(ExecutionEngine.java:59)
[error] org.neo4j.cypher.javacompat.ExecutionEngine.<init>(ExecutionEngine.java:43)
[error] org.neo4j.server.database.WrappedDatabase.<init>(WrappedDatabase.java:49)
[error] org.neo4j.server.database.WrappedDatabase$1.newDatabase(WrappedDatabase.java:41)
[error] org.neo4j.server.AbstractNeoServer.<init>(AbstractNeoServer.java:129)
[error] org.neo4j.server.CommunityNeoServer.<init>(CommunityNeoServer.java:62)
[error] org.neo4j.server.WrappingNeoServer.<init>(WrappingNeoServer.java:41)
[error] org.neo4j.server.WrappingNeoServer.<init>(WrappingNeoServer.java:36)
```

**Comment by chrisbeach** [comment]
Would appreciate if this issue was re-opened (to reflect the fact it's not yet resolved) - I'd like to know when Scala 2.11 compatibility is available.

**Comment by jonathancrosmer** [comment]
Agree with @chrisbeach.  Also, @cleishm those suggestions seem infeasible for users of Spring Data Neo4j.  SDN requires running in embedded mode to get any kind of reasonable performance.  Also it seems to require WrappingNeoServer to get back the webadmin support in Neo4j 2.x.  If WrappingNeoServer is removed or broken in a future release, I hope there is some other way to support webadmin with the embedded mode.
I realize SDN is not directly a part of Neo4j, but as long as Neo4j is offering an embedded mode, these are real issues.

**Comment by dwijnand** [comment]
I also agree this should be re-opened as I doubt this isn't on the roadmap at some stage.

For what it's work I had a look at what this might entail in August last year:

https://github.com/neo4j/neo4j/compare/2.3...beamly:bump-scala-to-2.11.2

The more interesting parts to make it "work" are:
- [making old compiler instances lazy](https://github.com/neo4j/neo4j/commit/a06e6c4fbee9a7378e66b030395445c0a2c18f1d)
- [ignoring some compatibility tests](https://github.com/neo4j/neo4j/commit/82e430d71dd9fabb157e2c6f9e0c810f14beb1fe) that can't work with a 2.11`scala-library` jar on the classpath

**Comment by richdougherty** [comment]
You may want to reconsider opening this now that Scala 2.11 has been out for a year. Most of the Scala ecosystem is now being [cross-built](http://www.scala-sbt.org/0.13/docs/Cross-Build.html) against both 2.10 and 2.11. Cross-building gives users a smooth migration path from 2.10 to 2.11.

In the next year or so, as [Scala 2.12 milestones come out](http://www.scala-lang.org/news/2.12.0-M1) and 2.12 is eventually released, we'll see more projects dropping 2.10 support completely and cross-building against Scala 2.11 and 2.12 instead. If libraries require 2.10 and don't support 2.11 then users will be stuck on 2.10 even as the rest of the ecosystem moves on. Users may find they can't use current versions of many libraries.

E.g. we're discussing dropping 2.10 support in [Play](https://github.com/playframework/playframework). We've been supporting both 2.10 and 2.11 for a year now. We'll probably drop support for 2.10 in the next version and support 2.11 and 2.12 instead. If users depend on libraries that require Scala 2.10 then they won't be able to upgrade to newer versions of Play.

So: +1. :)

**Comment by chrisvest** [comment]
Scala 2.11 support is being worked on in #4551
